api/services/request_body_parameter_validator.rb

```diff
@@ -0,0 +1,55 @@
+class RequestBodyParameterValidator
+  attr_reader :errors
+
+  def initialize(user_input)
+    @base_rent = user_input['base_rent']
+    @signed_on = user_input['signed_on']
+    @start_date = user_input['start_date']
+    @region = user_input['region']
+    @errors = {}
+    run
+  end
+
+  def errors?
+    !errors.empty?
+  end
+
+  private
+
```

Comment:
For readability I would use instance variables in order to access your instance methods
```suggestion
  attr_reader :base_rent, :signed_on, :start_date, :region, :error

```

And adapt the rest of the file